code/game/machinery/big_manipulator.dm

```diff
@@ -234,6 +390,68 @@
 /obj/machinery/big_manipulator/proc/finish_rotate_animation(backward)
 	animate(manipulator_hand, transform = matrix(180 * backward, MATRIX_ROTATE), working_speed*0.5)
 
+/obj/machinery/big_manipulator/proc/check_filter(obj/item/what_item)
+	var/filtered_obj = filter_obj?.resolve()
+	if(!istype(what_item, selected_type))
+		return
+	/// We use filter only on items. closets, humans and etc don't need filter check.
+	if(istype(what_item, /obj/item))
+		if((filtered_obj && !istype(what_item, filtered_obj)))
+			return FALSE
+	return TRUE
+
+/// Create dummy to force him use our item and then delete him.
+/obj/machinery/big_manipulator/proc/create_abstract_living()
+	var/mob/living/carbon/human/dummy/abstract_living = new /mob/living/carbon/human/dummy(get_turf(src))
+	abstract_living.alpha = 0
+	abstract_living.mouse_opacity = MOUSE_OPACITY_TRANSPARENT
+	ADD_TRAIT(abstract_living, TRAIT_UNDENSE, INNATE_TRAIT)
+	abstract_living.move_resist = INFINITY
+	abstract_living.invisibility = INVISIBILITY_ABSTRACT
+	abstract_living.real_name = abstract_living.name = name
+	abstract_living.mind_initialize()
+	return abstract_living
```

Comment:
it is taking like 80% of my will to not just yell at you guys for this
this is fucking insane, even if it didn't cause mob stacking we're lying about the mobs this could go wrong in like 50 different ways.

Reply:
for clarity I'm talkin bout maintainers shoulda trusted your gut on how sus this all is

Reply:
so i need to do a manipulator_act proc on any strcuctures and mobs or somthing

Reply:
Remember when you use your big manipulator to thank all the invisible humans that make it all possible.

Reply:
honestly I have no idea but this is asking for such trouble

Reply:
> Ask the contributor if it is using simulated ghost human dummies for human interactions or direct proc calls
> They're confused
->I pull out the profiler demonstrating the cost of human spawns compared to procs as needed.
> they laugh "It manipulates, sir!"
> Look at the code.
> It spawns human dummies

Reply:
Yeah, I should've trusted my gut.

Reply:
> so i need to do a manipulator_act proc on any strcuctures and mobs or somthing

if there is a solution to this it is to make the manipulator a mob, perhaps a robot. That'd require making the rest of the manip code work with it though, and would need to be done nicely. means mobs can exist in world properly rather then being a lie